content/code_snippets/displaying_images.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Displaying Images
+author: Adam Gerhant
+---
+
+This page will cover how to display an image with ImGui and canvas. The difference between the two methods is that canvas provides better drawing tools, while ImGui provides widgets which can be used interactively. It is possible to use both ImGui and canvas together for a combination of custom shapes and interactive controls.
+
+## Displaying with ImGui
+
+The main concept behind displaying an image with ImGui is to create a window with a clear background and no border. The image can then be displayed on this window.
+
+To start, create a clear window by using these flags in the begin function, as well as setting the position. This code should be in your Render() function.
+
+{{< highlight cpp "linenos=table" >}}
+ImGuiWindowFlags WindowFlags = ImGuiWindowFlags_NoDecoration | ImGuiWindowFlags_AlwaysAutoResize | ImGuiWindowFlags_NoFocusOnAppearing | ImGuiWindowFlags_NoBackground | ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoScrollbar;
+ImGui::SetNextWindowPos(ImVec2(1000.0f, 800.0f));
+if (!ImGuiBegin(menuTitle_.c_str(), &isWindowOpen_, WindowFlags))
+	{
+		 Early out if the window is collapsed, as an optimization.
+		ImGuiEnd();
+		return;
+}
+
+//code will be here
+
+ImGuiEnd();
+{{< / highlight >}}
+
+Since the image takes a few second to load from when the window is opened, it is best to always leave the window open and use an if statement to display your image.
```

Comment:
I tried moving the GetImGuiTex() to OnLoad but it doesnt work. I think the delay from displaying images is from the ImGui::Image function, not loading the texture. It probably is better to close the window instead of just not rendering anything, but I dont see any way of preventing the 1-2 second delay from displaying the image.